static/js/unread.js

```diff
@@ -458,11 +458,16 @@ function add_message_to_unread_mention_topics(message_id) {
 }
 
 function remove_message_from_unread_mention_topics(message_id) {
-    const message = message_store.get(message_id);
-    if (message.type !== "stream") {
+    const stream_id = unread_topic_counter.bucketer.reverse_lookup.get(message_id);
+    const per_stream_bucketer = unread_topic_counter.bucketer.get_bucket(stream_id);
+    if (!per_stream_bucketer) {
```

Comment:
Actually, is that right? Maybe `stream_id` would be the thing that's undefined then...

Reply:
It's a `get` call, so it doesn't crash:

```js
    get_bucket(bucket_key) {
        return this.key_to_bucket.get(bucket_key);
    }
```

I checked it in dev with PMs too.

Reply:
OK. I think I still prefer to check whether `stream_id` is undefined earlier; added that and made this step log something, since I think it'd be an invariant violation to get there.